src/common/Formatter.cc

```diff
@@ -426,40 +426,33 @@ void XMLFormatter::close_section()
     m_ss << "\n";
 }
 
-void XMLFormatter::dump_unsigned(const char *name, uint64_t u)
+template <class T>
+void XMLFormatter::add_value(const char *name, T val)
 {
   std::string e(name);
   std::transform(e.begin(), e.end(), e.begin(),
       [this](char c) { return this->to_lower_underscore(c); });
 
   print_spaces();
-  m_ss << "<" << e << ">" << u << "</" << e << ">";
+  m_ss.precision(std::numeric_limits<T>::max_digits10);
```

Comment:
nit, if we always want to `m_ss.precision(std::numeric_limits<T>::max_digits10)`, why not do so in the constructor?

Reply:
Yes, but since this is a template function, we can not guess `T` when we move that code to the constructor. Actually, we can, but this would be bad practice. In future, it is possible to have `dump_float()` and `dump_double()`  + `sump_long_double()` so I decide to make universal solution. Performance is not affected.